assets/src/util/fetchResult.ts

```diff
@@ -0,0 +1,21 @@
+interface Loading {
+  is_loading: true
+}
+
+interface Ok<T> {
+  ok: T
+}
+
+interface FetchError {
+  is_error: true
+}
+
+export type FetchResult<T> = Loading | Ok<T> | (Ok<T> & Loading) | FetchError
+
+export const isLoading = <T>(r: FetchResult<T>): r is Loading =>
+  "is_loading" in r && r.is_loading === true
```

Comment:
question/suggestion: would this work?
If it does, the `isFetchError` could also be changed

```suggestion
  r?.is_loading === true
```

Reply:
Unfortunately it doesn't - the `r?` only ensures that `r` is defined, so it still doesn't know if `is_loading` is a valid key and trying to use it in a comparison throws an error.

Reply:
Hmm, I was thinking that if `r = {}` then `r.is_loading` should be `undefined` and then when you do `undefined === true` should return `false` without errors, but if there's an error no worries.

Reply:
I guess `(r?.is_loading || undefined) === true` might work, but you're good to leave it as is lol.

Reply:
When trying that I still get:

```
src/util/fetchResult.ts:16:7 - error TS2339: Property 'is_loading' does not exist on type 'FetchResult<T>'.
  Property 'is_loading' does not exist on type 'FetchError'.
```

...so I think I'm just going to leave it as is and merge.